Unity-Technologies/ScriptableRenderPipeline
# URP/Shadergraph/XR GPU Instancing Tests

### Purpose of this PR
Currently, enabling "GPU Instancing" on shadergraph shaders breaks them. Sometimes on Desktop, sometimes in VR. This is not tested against yet.
This PR adds
- a test inside ShaderGraph testing with materials that enable "GPU Instancing"
- a test inside ShaderGraphUniversalStereo testing which is essentially the same but in XR

---
### Testing status

**Manual Tests**: What did you do?

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

_InstancedRendering test inside ShaderGraph project:_
![InstancedRendering](https://user-images.githubusercontent.com/2693840/66466712-72d9c300-ea83-11e9-9544-277412005a57.png)
Note that this fails when you disable SRP Batcher on the URPAsset:
![image](https://user-images.githubusercontent.com/2693840/66466789-9270eb80-ea83-11e9-8103-dcf5963adece.png)
Also note that even when it works the shaders have errors and a lot of warnings:
![image](https://user-images.githubusercontent.com/2693840/66466890-c3512080-ea83-11e9-87e2-d43a0a385494.png)

_InstancedRendering test inside ShaderGraphUniversalStereo project:_
![002_xr_InstancedRendering](https://user-images.githubusercontent.com/2693840/66466933-d237d300-ea83-11e9-8364-f7dbc5980e65.png)
Note that his currently fails as well on all XR rendering modes:
![image](https://user-images.githubusercontent.com/2693840/66466972-e5e33980-ea83-11e9-9120-cab16fb5ed82.png)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

---
### Overall Product Risks
**Technical Risk**: None

**Halo Effect**: Low (stuff needs to be fixed so these tests pass again)

---
### Comments to reviewers
I'd like to point out that I'd consider the underlying issues **high priority**. Those shader errors even prevent making builds.
